delivery/client.py

```diff
@@ -109,9 +133,123 @@ class DeliveryServiceClient:
     def __init__(
         self,
         routes: DeliveryServiceRoutes,
+        github_configs: collections.abc.Iterable[model.github.GithubConfig]=[],
+        cfg_factory: model.ConfigFactory | None=None,
```

Comment:
do we actually need to have the option to pass-in cfg-factory?

The reason I ask (+ would prefer not to) is the following. We will over time (hopefully) reduce the tight integration of delivery-svc-client to eventually mv it from cc-utils into open-component-model-org (and consume it from cc-utils as external dependency). Such that we can offer an easy-to-consume delivery-service-clients as part of ocm-gear.

cfg-factory is highly specific for our cicd environment. which is why we should avoid using it in code we eventually want to factor-out.

Reply:
At this point in time, I think the configuration factory is too tightly coupled with the ocm-gear, so that it cannot be removed easily. I suggest to postpone factoring-out of it until we have a suitable replacement, to prevent a list of workarounds until that point in time. Wdyt?

Reply:
agreed